user/enterprise/high-availability.md

```diff
@@ -4,11 +4,50 @@ layout: en_enterprise
 
 ---
 
-Travis CI Enterprise typically runs as a single container instance communicating with one or multiple workers, but we also offer a High Availability configuration so you can run your installation with redundancy. Particularly for customers with large-volume licenses, High Availability Mode is a helpful way to have additional stability.
+## Travis CI Enterprise 3.x 
+
+Travis CI Enterprise (TCIE) 3.x typically runs in **a Kubernetes cluster** with one or multiple workers. Services are split into several pods in the cluster. This enables a High Availability setup out of the box and you're able to customize based on your needs. 
+
+### Installing the Platform in High Availability Mode
+
+Please make sure to configure your [Kubernetes](https://kubernetes.io/) cluster with redundant pods for each service and secure appropriate resources and configuration (see below) to run it or choosing to self-host several services. You will need to modify several services configuration on your own to achieve High Availability adjusted to your requirements.
+
+Select one of the options depending on your planned setup, deployment and data backup policies and volume of users and build jobs. See also regular [installation instructions for TCIE 3.x](/user/enterprise/tcie-3.x-setting-up-travis-ci-enterprise#1-setting-up-enterprise-platform).
+
+#### Self hosting services for High Availability Mode 
+
+You may need to consider self-hosting of
+
+* [Redis](https://redis.io/), [RabbitMQ](https://www.rabbitmq.com/) and [Postgres](https://www.postgresql.org/) instances along with it's mirrors/redundant instances
+* Common Logs Target Location
+
+Once TCIE 3.x is installed, go to the Dashboard, Configs and format *self-hosted* options according to your planned setup. See example of self-hosted configuration for database access in TCIE 3.x below:
+
+![Self-hosted config example](/images/tcie-3.x-self-hosted-db.png)
+
+Similar configuration options are available for self-hosted logs, Insights, Redis and Rabbit MQ. The self-hosted solution for these services allows you to fully deploy and configure their High Availability compliant setup as per your requirements.
+
+#### Utilizing services in Kubernetes cluster
+
+Other option is to modify configuration for aforementioned services (PostgreSQL, Redis and RabbitMQ) taking into account service specific details. These details can be found in the documentation for the tools. Since there are multiple possible variations, we suggest only general guidelines which should be treated as a starting point for developing your specific High Availability configuration.
```

Comment:
**Another** option is to modify the configuration for the aforementioned services (PostgreSQL, Redis**,** and RabbitMQ) taking into account service specific details. These details can be found in the documentation for the tools. Since there are multiple possible variations, we suggest only general guidelines **that** should be treated as a starting point for developing your specific High Availability configuration.

Reply:
applied in subsequent commit